testcommon/testCallGraphUtils.go

```diff
@@ -0,0 +1,295 @@
+package testcommon
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ElrondNetwork/arwen-wasm-vm/v1_4/arwen"
+	"github.com/ElrondNetwork/arwen-wasm-vm/v1_4/arwen/elrondapi"
+	mock "github.com/ElrondNetwork/arwen-wasm-vm/v1_4/mock/context"
+	"github.com/ElrondNetwork/elrond-vm-common/txDataBuilder"
+	"github.com/stretchr/testify/require"
+)
+
+// TestReturnDataSuffix -
+var TestReturnDataSuffix = "_returnData"
+
+// TestCallbackPrefix -
+var TestCallbackPrefix = "callback_"
+
+// TestContextCallbackFunction -
+var TestContextCallbackFunction = "contextCallback"
+
+// CreateMockContractsFromAsyncTestCallGraph creates the contracts
+// with functions that reflect the behavior specified by the call graph
+func CreateMockContractsFromAsyncTestCallGraph(callGraph *TestCallGraph, testConfig *TestConfig) []MockTestSmartContract {
+	contracts := make(map[string]*MockTestSmartContract)
```

Comment:
this should be split in smaller, easier to follow/debug functions

Reply:
It's already being refactored on a separate branch.

Reply:
great